### ISSUE TITLE
[HOLD][$1000] IOU - digit appears cutoff for a moment when entering a digit before decimal point

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#17801]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app on android chrome
2. Click on plus
3. Click on request money/send money
4. Write any amount of digits, add dot and now get the cursor before dot
5. Insert any digit (2 and 5 has most amount of cut and lag) and observe that digit is cut off for few seconds and dot is also not visible for that moment

## Expected Result:
 App should display all the digits and dot without cutting it off like is does on android app and all other platforms
## Actual Result: 
When trying to add new digit before dot, app cuts off the digit for few seconds on android chrome
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/233728081-ff5baaa3-1aa0-478b-a66c-47ae1acee18c.mp4

https://user-images.githubusercontent.com/43996225/233728103-c05beee0-a4e7-461e-a378-90e874d4fa29.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1682066050918369

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019e5fd2a492a92ff5</li>
        <li>Upwork Job ID: 1650216918652129280</li>
        <li>Last Price Increase: 2023-04-23</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @strepanier03 (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by strepanier03** [comment]
I am able to recreate this behavior. 

Tested with Google Pixel 6 Pro on m/web Chrome staging.new.expensify.com.

It is most pronounced with the "5" digit in my opinion but it is still fairly quick to correct placement. 

I think it's a UX improvement but not a majorly huge one because it's just a few fractions of a second off and isn't overly noticeable.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~019e5fd2a492a92ff5

**Comment by MelvinBot** [comment]
Current assignee @strepanier03 is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @parasharrajat (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @aldo-expensify (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by strepanier03** [comment]
If you disagree with this being fixed let me know!

**Comment by aldo-expensify** [comment]
It could be a performance issue that is only visible in slow phones 🤷 , it doesn't look great, so I think it is fine if we leave it open for investigation.

I agree with it having low priority.

**Comment by strepanier03** [comment]
Thank you for weighing in @aldo-expensify and @parasharrajat - Leaving it open and external. 

I am going to be out of office April 26-May 3 but I don't expect my action will be needed on this GH before then so I don't plan to unassign this. 

If something is needed from a BZ team member please post in #expensify-open-source and someone can help out.

**Comment by mountiny** [comment]
maybe @narefyev91 would be interested in this one

**Comment by narefyev91** [comment]
Hi I'm Nicolay from Callstack - expert contributor group - will start investigation in this area

**Comment by aldo-expensify** [comment]
Thanks @narefyev91 !

**Comment by MelvinBot** [comment]
📣 @narefyev91 You have been assigned to this job by @mountiny!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by narefyev91** [comment]
My updates here - still investigating - was able to reproduce on real device - not only with dot, but with randomly moving cursor and typing

**Comment by narefyev91** [comment]
Updates from my side - indicate a root case of the issue - trying to find a solution to fix - will provide internal proposal as soon as possible

**Comment by narefyev91** [comment]
Posted local proposal for review - will add it here as soon as i get approve

**Comment by narefyev91** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Resize of dynamic input happens after number already presented inside input component

### What is the root cause of that problem?
The root cause of problem - is sequence of how we manipulate with state in lifecycle and how BaseTextInput component with props `autoGrow` is working.
The sequence is this: user click on NumPad -> we send new **value** prop -> componentDidUpdate fires to set value from props -> value is showing in input -> Text component(which is hidden for calculating width) onLayout fires -> set new width -> width applies to input.
The issue here that we should not set new value inside input - until we change width of the input - because width is dynamic and should be calculated based on value.

### What changes do you think we should make in order to solve the problem?
All changes should be done only if **autoGrow is true**
1) we need to add local value - based on it we will calculate width of dynamic input
2) value for input should be set only when width will be calculated and set after onLayout event from hidden Text component
Most of the changes will be done inside componentDidUpdate:
```
    componentDidUpdate(prevProps, prevState) {
        // Activate or deactivate the label when value is changed programmatically from outside
        const inputValue = _.isUndefined(this.props.value) ? this.input.value : this.props.value;
        if ((_.isUndefined(inputValue) || this.state.value === inputValue) && _.isEqual(prevProps.selection, this.props.selection)) {
            return;
        }

        if (this.props.autoGrow) {
            if (inputValue !== this.state.localValue) {
                this.setState({localValue: inputValue});
            }

            if (prevState.textInputWidth === this.state.textInputWidth) {
                return;
            }
        }
```
Hidden Text will be changes from this.state.value -> this.state.localValue
```
            {this.props.autoGrow && (

                // Add +2 to width so that the first digit of amount do not cut off on mWeb - https://github.com/Expensify/App/issues/8158.
                <Text
                    style={[...this.props.inputStyle, styles.hiddenElementOutsideOfWindow, styles.visibilityHidden]}
                    onLayout={e => this.setState({textInputWidth: e.nativeEvent.layout.width + 2})}
                >
                    {this.state.localValue || this.props.placeholder}
                </Text>
            )}
```

### What alternative solutions did you explore?

cc @strepanier03 @aldo-expensify @parasharrajat

**Comment by aldo-expensify** [comment]
@narefyev91 please start working in a PR and then we can test if it causes delays for the user.

**Comment by strepanier03** [comment]
All good Melvin, PR was submitted and will be reviewed shortly.

**Comment by strepanier03** [comment]
Discussion is ongoing on the PR and it looks like we're making progress.

**Comment by strepanier03** [comment]
Bumped on PR.

**Comment by strepanier03** [comment]
Update on PR, will be reviewed within the next few hours so we're still moving along here.

**Comment by strepanier03** [comment]
PR discussion ongoing and moving forward, looking okay here.

**Comment by strepanier03** [comment]
Bumped on [PR](https://github.com/Expensify/App/pull/18579#issuecomment-1568834896) so things don't get stale.

**Comment by strepanier03** [comment]
@parasharrajat is testing today, will follow up tomorrow.

**Comment by strepanier03** [comment]
Some discussion on the [PR](https://github.com/Expensify/App/pull/18579) so I bumped C+ and Eng for some thoughts.

**Comment by strepanier03** [comment]
Moving this to weekly since it's on hold.

**Comment by strepanier03** [comment]
@zanyrenney - I will be on sabbatical before this comes off hold so I need to hand it over. Thank you for taking it on. 

For now it's just on hold to be monitored, it just went to hold so it doesn't need a check for another week unless something changes and you get pinged.

**Comment by zanyrenney** [comment]
No worries @strepanier03 sounds good. I'll keep an eye on this for now.

**Comment by zanyrenney** [comment]
Hey, I am heading OOO shortly, so assigning another bug0 team member to help watch over this issue whilst I am OOO.

**Comment by johncschuster** [comment]
I think the `Daily` label was automatically added when the `Bug` label was reapplied. I'm going to downgrade this back to `Weekly` while it's still on hold.

**Comment by johncschuster** [comment]
Bumping back to `Weekly`

**Comment by dhanashree-sawant** [comment]
Yes it is still reproducible, although animation is slighter faster so lesser noticeable.

**Comment by johncschuster** [comment]
Thanks for confirming that, @dhanashree-sawant! Looks like https://github.com/Expensify/App/pull/20186 (and the regression it introduced) is now resolved. Let's get back to this one!

**Comment by narefyev91** [comment]
@johncschuster Hey! https://github.com/Expensify/App/pull/18579#issuecomment-1664509209 we already have a draft PR which will make behaviour almost gone - but we also still need some additional fixes on native side https://github.com/Expensify/App/pull/18579#issuecomment-1665138896

**Comment by johncschuster** [comment]
Oh sweet! Thanks for those updates!

**Comment by johncschuster** [comment]
Looks like we're still waiting on https://github.com/Expensify/App/pull/18579, right @narefyev91?

**Comment by narefyev91** [comment]
@johncschuster Hey! We waiting this one https://github.com/Expensify/App/pull/18579#issuecomment-1664509209 + fixes on native side i think

**Comment by johncschuster** [comment]
Sorry I missed your comment, @narefyev91! Thanks for the update!

**Comment by johncschuster** [comment]
Looks like we're still waiting on https://github.com/Expensify/App/pull/18579#issuecomment-1664509209, which is currently on hold.

**Comment by BlameGPT** [comment]
t

**Comment by BlameGPT** [comment]
Contributor details
Your Expensify account email: rushat222@gmail.com
Upwork Profile Link: http://upwork.com/freelancers/~018e905fle69c8096d

**Comment by BlameGPT** [comment]
Contributor details
Your Expensify account email: rushat222@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01990b51632551e540

**Comment by strepanier03** [comment]
Still on hold.
